TypeCobol.LanguageServer/WorkspaceProject.cs

```diff
@@ -0,0 +1,250 @@
+        {
+            if (!IsSourceProviderChanged(copyFolders))
+            {
+                return false;
+            }
+            // Now set new copy Directories
+            Project.SourceFileProvider.Clear();
+            foreach (var copyFolder in copyFolders)
+            {
+                Project.SourceFileProvider.AddLocalDirectoryLibrary(copyFolder, false,
+                    new[] { ".cpy" }, configuration.Format.Encoding,
+                    configuration.Format.EndOfLineDelimiter, configuration.Format.FixedLineLength);
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Try to get DocumentContext in this Workspace Project, it the given Uri instance exists
+        /// </summary>
+        /// <param name="fileUri">The Uri instance to get the DocumentContext instance</param>
+        /// <param name="openedDocumentContext">[out] the DocumentContext instance if the Uri exists</param>
+        /// <returns>true if a DocumentContext instance has been found, false otherwise</returns>
+        internal bool TryGetOpenedDocumentContext(Uri fileUri, out DocumentContext openedDocumentContext)
+        {
+            return _openedDocuments.TryGetValue(fileUri, out openedDocumentContext);
+        }
+
+        /// <summary>
+        /// Try to get a DocumentContext associated to the given CompilationUnit
+        /// </summary>
+        /// <param name="cunit">The CompilationUnit</param>
+        /// <param name="openedDocumentContext"></param>
+        /// <returns>true if the DocumentContext has been found, false otherwise</returns>
+        internal bool TryGetOpenedDocumentContext(CompilationUnit cunit, out DocumentContext openedDocumentContext)
+        {
+            openedDocumentContext = null;
+            if (cunit == null)
+                return false;
+            Uri fileUri = this._openedDocuments.Keys.FirstOrDefault(k => k.LocalPath.Contains(cunit.TextSourceInfo.Name));
+            if (fileUri != null)
+            {
+                openedDocumentContext = this._openedDocuments[fileUri];
+                return true;
+            }
+            return false;
+        }
+        
+        /// <summary>
+        /// Indicates whether this workspace project has opened documents or not.
+        /// </summary>
+        internal bool IsEmpty
+        {
+            get 
+            {
+                return _openedDocuments.IsEmpty;
+            }            
+        }
+
+        /// <summary>
+        /// Refresh all documents of this WorkspaceProject instance
+        /// </summary>
+        internal void DoRefreshOpenedFiles(Workspace workspace)
+        {
+            this.Project.ClearImportedCompilationDocumentsCache();
+            foreach (var contextEntry in _openedDocuments)
+            {
+                DocumentContext docContext = contextEntry.Value;
+                docContext.FileCompiler.CompilationProject = Project;
```

Comment:
This because the CompilationProject to the default WorkspaceProjet is always recreate at each global DidConfigurationChange notification, so it was done like that, but ok I am going to remove this assignment here and do it in ReplaceCompilationProject.